src/hardware/disney.cpp

```diff
@@ -27,49 +27,52 @@
 #include "pic.h"
 #include "setup.h"
 
-#define DISNEY_BASE 0x0378
+// Disney Sound Source Constants
+constexpr uint16_t DS_PORT_BASE = 0x0378;
+constexpr uint8_t DS_BUF_SAMPLES = 128;
 
-#define DISNEY_SIZE 128
+constexpr uint8_t DS_ACKNOWLEDGE_BIT = 0b0100'0000;  // 0x40
+constexpr uint8_t DS_INTERRUPT_MASK = 0b1111'1011;   // ~0x4
+constexpr uint8_t DS_INIT_STATUS_BITS = 0b1000'0100; // 0x84
+constexpr uint8_t DS_PARALLEL_IRQ_BIT = 0b0001'0000; // 0x10
+constexpr uint8_t DS_PIN_9_BIT = 0b1000'0000;        // 0x80
 
-constexpr uint8_t DISNEY_INIT_STATUS = 0b1000'0100; // 0x84
+enum DS_STATE { IDLE, RUNNING, FINISH, ANALYZING };
```

Comment:
Previously these values had associated numbers - are you 100% sure it's not needed any more? The code suggests that these values were meaningful.

Reply:
Can you point to the code that indicates the numerical values were actually used?  

My analysis showed the values are independent states and there is no code that numerically uses their prior 'values'. 

For example, here is everwhere the `state` value is referenced:

``` c++

disney.state = DS_IDLE;                                      disney.state = DS_IDLE;                              disney.state = DS_RUNNING;                   switch(disney.state) {                                               disney.state = DS_ANALYZING;                                 disney.state=DS_IDLE;                                disney.state = DS_FINISH;                                                         if(disney.state != DS_RUNNING) {                                     if(disney.state != DS_RUNNING) {                                                                          if(disney.state != DS_RUNNING) {
                        if(disney.state != DS_RUNNING) {
```

No addition, subtraction, greater or less than comparisons, and so on.